### PR TITLE
Added more exception handling in Az.HealthcareApis

### PR DESCRIPTION
**Pull request by poadhika**
## Description

Added more exception handling in Az.HealthcareApis

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/HealthcareApis/HealthcareApis/Properties/Resources.resx [hunk 1]
```diff
@@ -141,4 +141,10 @@
   <data name="invalidAccessPolicyObjectIdMessage" xml:space="preserve">
     <value>At least one of the supplied access policy object IDs is invalid. Object ID must be a valid GUID.</value>
   </data>
+  <data name="keyNotFoundExceptionMessage" xml:space="preserve">
+    <value>Object Id couldn't be retrieved from the current context. Please specify at least one  accesspolicy object id</value>
```
Comment (erich-wang) [inline src/HealthcareApis/HealthcareApis/Properties/Resources.resx]: nit: remove duplicate white spaces  after `one`

File: src/HealthcareApis/HealthcareApis/Models/PSHealthcareApisService.cs [hunk 2]
```diff
@@ -73,7 +73,7 @@ private static string GetKindValue(Kind kind)
             switch (kind)
             {
                 case Management.HealthcareApis.Models.Kind.Fhir:
-                    return "fhir";
+                    return "fhir-Stu3";
```
Comment (erich-wang) [inline src/HealthcareApis/HealthcareApis/Models/PSHealthcareApisService.cs]: Is this change on purpose?

Comment (poadhika) [reply src/HealthcareApis/HealthcareApis/Models/PSHealthcareApisService.cs]: yes

**Comment by erich-wang** [comment]
@poadhika , please fix test error.

**Comment by poadhika** [comment]
@erich-wang I have fixed the text errors. Please take a look
